### PR TITLE
driver: sdhc: ESP32 pinctrl config for SDHC

### PR DESCRIPTION
**Pull request by raffarost**
Additional LL files for SDHC driver, as well as pinctrl configuration

### CODE REVIEW

File: components/soc/esp32/include/soc/sdmmc_struct.h [hunk 1]
```diff
@@ -1,5 +1,9 @@
 /*
+<<<<<<< HEAD
  * SPDX-FileCopyrightText: 2015-2022 Espressif Systems (Shanghai) CO LTD
+=======
+ * SPDX-FileCopyrightText: 2015-2023 Espressif Systems (Shanghai) CO LTD
+>>>>>>> 952499f843 (Preliminary version of files for SDIO driver)
```
Comment (sylvioalves) [inline components/soc/esp32/include/soc/sdmmc_struct.h:6]: Please, check this diff.

Comment (raffarost) [reply components/soc/esp32/include/soc/sdmmc_struct.h]: Fixed. Thanks

**Review by LucasTambor — APPROVED** [review]

**Review by marekmatej — APPROVED** [review]

**Comment by novatia** [comment]
Hello everyone,
I`m new to zephyr.
I've identified a bug in the current version of West that affects the update process, and I'd like to share the steps to reproduce it, as well as how this pull request resolves the issue.

As stated in zephyr getting started guide follow steps until "west update" command (I'm using windows). 
West fails reporting this error:

ERROR: update failed for project hal_espressif

And its details:

=== updating hal_espressif (modules/hal/espressif):
--- hal_espressif: fetching, need revision e554857b07d54e2152c6f1a9b39da6d6c13fe7f7

I solved getting the pr commit as follows:

git clone https://github.com/zephyrproject-rtos/hal_espressif %HOMEPATH%\zephyrproject\modules\hal\espressif
git fetch origin pull/276/head:pr_276
git checkout pr_276

When running again west update I got a Warning but following next steps on the getting started guide now seems working.

Best regards.

**Comment by raffarost** [comment]
Hi novatia,

You're right about using this PR with the latest pull o Zephyr. It should have been integrated already.
I'll check on that asap.

Thank you for your feedback

**Pull request merged**